plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/SystemTableAwareAccessControl.java

```diff
@@ -87,4 +92,13 @@ public void checkCanSelectFromColumns(ConnectorSecurityContext context, SchemaTa
 
         delegate.checkCanSelectFromColumns(context, tableName, columnNames);
     }
+
+    private Optional<SchemaTableName> getSourceTableNameFromSystemTable(SchemaTableName tableName)
```

Comment:
can we not copy-paste logic?

Reply:
You at very least can keep the static method, just parameterized it with `Set<SystemTableProvider>`. And call in two places.

Reply:
done